x/concentrated-liquidity/pool.go

```diff
@@ -164,42 +166,43 @@ func (k Keeper) CalcOutAmtGivenIn(ctx sdk.Context,
 			swapState.amountSpecifiedRemaining,
 			zeroForOne,
 		)
+		swapState.sqrtPrice = sqrtPrice
 
 		swapState.amountSpecifiedRemaining = swapState.amountSpecifiedRemaining.Sub(amountIn)
 		swapState.amountCalculated = swapState.amountCalculated.Add(amountOut)
 
 		// if we have moved to the next tick,
-		if nextSqrtPrice.Equal(sqrtPrice) {
+		if nextSqrtPrice.Equal(swapState.sqrtPrice) {
```

Comment:
I think this can stay the same? (use sqrtPrice)

Reply:
they are synonymous now since we we now set swapState.sqrtPrice = sqrtPrice but I will change it back!